.github/workflows/ci.yml

```diff
@@ -0,0 +1,285 @@
+        uses: actions/checkout@v2
+      - name: Cache node
+        uses: actions/cache@v1
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ matrix.NODEJS_VERSION }}-${{ hashFiles('**/lektor/admin/package-lock.json') }}
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.NODEJS_VERSION }}
+      - name: Install node dependencies
+        run: |
+          cd lektor/admin/
+          npm install .
+          npm run webpack
+      - name: Show node environment
+        run: |
+          node --version
+          npm --version
+
+  win-node:
+    name: Win node${{ matrix.NODEJS_VERSION }} tests
+    runs-on: windows-latest
+    needs: lint-js
+    env:
+      CI: True
+      NODEJS_VERSION: ${{ matrix.NODEJS_VERSION }}
+    strategy:
+      fail-fast: false 
+      matrix:
+        NODEJS_VERSION: ['10', '12', '13']
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+      - name: Cache node
+        uses: actions/cache@v1
+        with:
+          path: ~\AppData\Roaming\npm-cache
+          key: ${{ runner.os }}-node-${{ matrix.NODEJS_VERSION }}-${{ hashFiles('**\lektor\admin\package-lock.json') }}
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.NODEJS_VERSION }}
+      - name: Install node dependencies
+        run: |
+          cd lektor/admin/
+          npm install .
+          npm run webpack
+      - name: Show node environment
+        run: |
+          node --version
+          npm --version
+
+  ############################################################################
+  # Python tests
+  ############################################################################
+  linux-py:
+    name: Linux py${{ matrix.PYTHON_VERSION }} tests
+    runs-on: ubuntu-latest
+    needs: lint-py
+    env:
+      CI: True
+      PYTHON_VERSION: ${{ matrix.PYTHON_VERSION }}
+    strategy:
+      fail-fast: false 
+      matrix:
+        PYTHON_VERSION: ['2.7', '3.6', '3.8']
+    steps:
+      - name: Checkout branch
+        uses: actions/checkout@v2
+      - name: Cache pip
+        uses: actions/cache@v1
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-py-${{ matrix.PYTHON_VERSION }}-pip-${{ hashFiles('**/setup.py') }}
+      - name: Setup python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.PYTHON_VERSION }}
+      - name: Install python dependencies
+        run: |
+          python -m pip install --editable .[test]
+          python -m pip install codecov
+      - name: Show python environment
+        run: |
+          python --version
+          python -m pip list
+      - name: Run python tests
+        run: pytest . --tb=long -svv --cov=lektor
+      - name: Publish coverage
+        if: matrix.PYTHON_VERSION == '3.8'
+        run: codecov -t 39974034-91d8-48d1-9698-de48e0667a09
```

Comment:
I don't think we should expose the token like this unless we really have to. There are ways to set environment variable in a more secret way. (https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets)

If we keep this here I think forks will report their coverage centrally as well.

Reply:
Secrets are not passed to PRs from forks AFAIK

https://github.community/t5/GitHub-Actions/Make-secrets-available-to-builds-of-forks/td-p/30678